src/app/item-popup/Plug.tsx

```diff
@@ -44,30 +47,126 @@ export default function Plug({
   const selectable = socketInfo.plugOptions.length > 1;
 
   return (
     <div
       key={plug.plugDef.hash}
       className={clsx('socket-container', {
         disabled: !plug.enabled,
-        notChosen: !plugged,
         selectable,
-        // This has been selected by the user but isn't the original plugged item
-        selected: socketInfo.actuallyPlugged && plugged,
-        // Another plug was selected by the user
-        notSelected: socketInfo.actuallyPlugged && !plugged && plug === socketInfo.actuallyPlugged,
         notIntrinsic: !itemCategories.includes(ItemCategoryHashes.WeaponModsIntrinsic),
-        cannotRoll: plug.cannotCurrentlyRoll,
       })}
       onClick={hasMenu || selectable ? doClick : undefined}
     >
-      <PressTip tooltip={tooltip}>{contents}</PressTip>
+      {socketInfo.isReusable ? (
+        <PerkCircleWithTooltip
+          item={item}
+          plug={plug}
+          wishlistRoll={wishlistRoll}
+          socketInfo={socketInfo}
+        />
+      ) : (
+        <>
+          <PressTip tooltip={tooltip}>{contents}</PressTip>
+          {wishlistRoll?.wishListPerks.has(plug.plugDef.hash) && (
+            <AppIcon
+              className="thumbs-up"
+              icon={thumbsUpIcon}
+              title={t('WishListRoll.BestRatedTip')}
+            />
+          )}
+        </>
+      )}
+    </div>
+  );
+}
+
+/**
+ * a perk circle and its associated thumbs up or lack thereof.
+ * this belongs inside an element with a css position, so thumbs up can position itself.
+ */
+export function PerkCircleWithTooltip({
+  item,
+  plug,
+  socketInfo,
+  wishlistRoll,
+}: {
+  item: DimItem;
+  plug: DimPlug;
+  socketInfo: DimSocket;
+  wishlistRoll?: InventoryWishListRoll;
+}) {
+  const plugged = plug === socketInfo.plugged;
+
+  // Another plug was selected by the user
+  const notSelected = socketInfo.actuallyPlugged && !plugged && plug === socketInfo.actuallyPlugged;
+  // This has been selected by the user but isn't the original plugged item
+  const selected = socketInfo.actuallyPlugged && plugged;
+  const cannotRoll = plug.cannotCurrentlyRoll;
+
+  const tooltip = () => <DimPlugTooltip item={item} plug={plug} wishlistRoll={wishlistRoll} />;
+  return (
+    <>
+      <PressTip tooltip={tooltip}>
+        <PerkCircle plug={plug} {...{ plugged, notSelected, selected, cannotRoll }} />
+      </PressTip>
       {wishlistRoll?.wishListPerks.has(plug.plugDef.hash) && (
         <AppIcon className="thumbs-up" icon={thumbsUpIcon} title={t('WishListRoll.BestRatedTip')} />
       )}
-    </div>
+    </>
+  );
+}
+
+type PlugStatuses = {
+  plugged?: boolean;
+  selected?: boolean;
+  cannotRoll?: boolean;
+  notSelected?: boolean;
+};
+
+/** an encircled perk image */
+function PerkCircle({
+  plug,
+  className,
+  ...plugStatuses
+}: {
+  plug: DimPlug;
+  className?: string;
+} & PlugStatuses) {
+  const enhanced = isEnhancedPerk(plug);
+  const statusClasses = clsx(plugStatuses);
```

Comment:
Note that by passing these in as an object you're not going to get the benefit of our Babel optimize-clsx plugin which turns these into a bunch of conditionals instead of a function call.